layers.go

```diff
@@ -677,7 +691,7 @@ func (r *layerStore) Unmount(id string) error {
 		layer.MountPoint = ""
 		err = r.Save()
 	}
-	return err
+	return false, err
```

Comment:
If `err` is not `nil`, this probably needs to be `true`, since we're leaving the recorded `MountCount` unchanged.